### PR TITLE
Delete tenants from Tenant service when deprovisioning user

### PR DESCRIPTION
**Pull request by alexeykazakov**
Part of https://github.com/openshiftio/openshift.io/issues/2393

### CODE REVIEW

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-484-2`

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-484-3`

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-484-4`

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-484-5`

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-484-6`

File: account/service/tenant.go [hunk 1]
```diff
@@ -39,15 +52,68 @@ func InitTenant(ctx context.Context, config tenantConfig) error {
...
+	}
+	defer rest.CloseResponse(response)
+
+	if response.StatusCode != http.StatusNoContent {
+		log.Error(ctx, map[string]interface{}{
+			"identity_id":     identityID.String(),
+			"response_status": response.Status,
+			"response_body":   rest.ReadBody(response.Body),
+		}, "unable to delete tenants")
+		return errors.New("unable to delete tenant")
```
Comment (sbose78) [inline account/service/tenant.go]: What's the difference between this and `errors.NewInternalError...()` ?

File: configuration/configuration.go [hunk 2]
```diff
@@ -584,6 +584,18 @@ func (c *ConfigurationData) DefaultConfigurationError() error {
 	return c.defaultConfigurationError
 }
 
+// GetAuthServiceUrl returns Auth Service URL
+func (c *ConfigurationData) GetAuthServiceURL() string {
```
Comment (sbose78) [inline configuration/configuration.go]: finally! :) 
Explicit is good.

File: controller/users.go [hunk 3]
```diff
@@ -756,8 +758,23 @@ func (c *UsersController) UpdateByServiceAccount(ctx *app.UpdateByServiceAccount
...
 		}
+		if c.tenantService != nil {
+			// Delete tenant
+			err = c.tenantService.Delete(ctx, identity.ID)
+			if err != nil {
+				log.Error(ctx, map[string]interface{}{
+					"err":         err,
+					"identity_id": ctx.ID,
+				}, "unable to delete tenant when deprovisioning user")
+				// Log the error and proceed
```
Comment (sbose78) [inline controller/users.go]: Good! The workflow shouldn't hit a wall if for some reason the `delete tenant` fails.

**Comment by sbose78** [comment]
Looks good, I have a change proposal https://github.com/alexeykazakov/fabric8-auth/pull/1

**Comment by alexeykazakov** [comment]
> Looks good, I have a change proposal alexeykazakov#1

Good catch.

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-484-7`

**Comment by alexeykazakov** [comment]
> Looks good, but makes me wonder if Deprovision should be a different endpoint altogether instead of clubbing it in update because clubbing it inside update exposes implementation detail.

Maybe.. But probably it's a bit late to change this API at this point.

**Comment by alexeykazakov** [comment]
[test]

**Pull request merged**